src/web/app/src/components/Posts/Post.tsx

```diff
@@ -327,6 +337,51 @@ function handleZoom(e: MouseEvent) {
     e.preventDefault();
     zoomInImage(e.target);
   }
+
+  // if the user clicks the copy button
+  if (e.target instanceof HTMLElement && e.target.closest('.copyCodeBtn')) {
+    e.preventDefault();
+    copyCode(e.target);
+  }
+}
+
+function createCopyButton(e: MouseEvent) {
+  if (
+    e.target instanceof HTMLElement &&
+    e.target.tagName === 'CODE' &&
+    e.target.parentElement?.tagName === 'PRE'
```

Comment:
Could make this a function

Reply:
With all the changes you suggested it complains that `e` is possibly null. I'm not sure how to handle it

Reply:
`e?.target`